### DISCUSSION TITLE
Adding comment from a rule

### DISCUSSION THREAD

**Discussion by marnheus** [tfsaggregator/aggregator-cli#259] [Q&A]
Hello is there a way to add a comment to a work item during the execution of a rule?

Basically I'd like to explain why a state was reverted by the rule, but I don't see anything like that in the documentation.

Thanks.

**Comment by giuliov** [comment]
I am not sure to understand where you want to see the comment. I know two options.

1. the purpose of the `Reason` field is to explain why the work item landed on a particular state,
in the old XML Process it was set automatically, but should work with Inherited Process
2. the `History` field is a list of comments: what you set will be _appended_ to the list

HTH

**Reply by marnheus** [reply]
Basically, I will revert a state change because of some rule. What I'd like is to add a comment (that would be visible in the discussion part) to explain what happened.
So in order to do that, adding something to history would work?

**Reply by giuliov** [reply]
The `History` field should be it.
